synapse/models/infotech.py

```diff
@@ -1937,6 +2039,24 @@ def getModelDefs(self):
                     }),
                 )),
                 ('it:cmd', {}, ()),
+                ('it:cmd:session', {}, (
+                    ('host', ('it:host', {}), {
+                        'doc': 'The host where the command line session was executed.'}),
+                    ('period', ('ival', {}), {
+                        'doc': 'The period over which the command line session was running.'}),
+                    ('file', ('file:bytes', {}), {
+                        'doc': 'The file containing the command history such as a .bash_history file.'}),
+                )),
+                ('it:cmd:history', {}, (
+                    ('cmd', ('it:cmd', {}), {
+                        'doc': 'The command that was executed.'}),
+                    ('session', ('it:cmd:session', {}), {
+                        'doc': 'The session that contains this history entry.'}),
+                    ('time', ('time', {}), {
```

Comment:
re :time, how would you deconflict (or not) commands with known time run w/in an it:cmd:session on an it:host vs it:exec:proc (:cmd) run on the same host? I understand that an it:cmd:session allows us to group commands that might appear "independent" (interleaved w/other commands) as it:exec:proc's. But this seems to leave us with and either/or way to model command execution, or require us to model some commands as both it:exec:proc and it:cmd:history / session?